x-pack/legacy/plugins/siem/server/lib/pinned_event/saved_object.ts

```diff
@@ -97,21 +97,21 @@ export class PinnedEvent {
     eventId: string,
     timelineId: string | null
   ): Promise<PinnedEventSavedObject | null> {
-    let timelineVersionSavedObject = null;
     try {
-      if (timelineId == null) {
-        const timelineResult = convertSavedObjectToSavedTimeline(
-          await this.libs.savedObjects
-            .getScopedSavedObjectsClient(request[internalFrameworkRequest])
-            .create(
-              timelineSavedObjectType,
-              pickSavedTimeline(null, {}, request[internalFrameworkRequest].auth || null)
-            )
-        );
-        timelineId = timelineResult.savedObjectId;
-        timelineVersionSavedObject = timelineResult.version;
-      }
       if (pinnedEventId == null) {
+        let timelineVersionSavedObject = null;
```

Comment:
optional: this `let` looks like it can be made `const` with something like (untested):

```
        const timelineVersionSavedObject = await (async () => {
          if (timelineId == null) {
            const timelineResult = convertSavedObjectToSavedTimeline(
              await this.libs.savedObjects
                .getScopedSavedObjectsClient(request[internalFrameworkRequest])
                .create(
                  timelineSavedObjectType,
                  pickSavedTimeline(null, {}, request[internalFrameworkRequest].auth || null)
                )
            );
            timelineId = timelineResult.savedObjectId;
            return timelineResult.version;
          }
        })();
```